### PR TITLE
Copy notebook

### PR DESCRIPTION
**Pull request by twentyse7en**
Copy notebook cells directly from GUI.

## Contributor Checklist:

* [x] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [ ] I have added/updated documentation for all user-facing changes
* [x] I have added/updated test cases
* [ ] I have included the [rebuilt production build of the client](https://libertem.github.io/LiberTEM/contributing.html?#building-the-client) (only if
changes were made to the GUI)

### CODE REVIEW

**Comment by twentyse7en** [comment]
Where would be the ideal place for the modal trigger button?

**Comment by sk1p** [comment]
> Where would be the ideal place for the modal trigger button?

Hmm - I think downloading results, notebooks and displaying/copying script cells are all related to one another. Maybe the download button can be converted to an "export" or "share" button? Possibly, it could all fit into a single modal, having multiple tabs - one for downloading results, one for notebooks, and one for showing code / copying / ... what do you think? I'm open to other ideas, too!

**Comment by twentyse7en** [comment]
Are you suggesting something like this by "multiple tabs"?

image

**Comment by sk1p** [comment]
> Are you suggesting something like this by "multiple tabs"?
> 
> image

Yes, that looks nice!

**Comment by twentyse7en** [comment]
`test_copy` is almost identical to the test for downloading the notebook, I don't understand why the status is 404. Can you please look @sk1p

**Comment by sk1p** [comment]
> `test_copy` is almost identical to the test for downloading the notebook, I don't understand why the status for response is 404. Can you please look @sk1p

I'll have a look

**Comment by sk1p** [comment]
I think the problem is the following:

```
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: text/html; charset=utf-8', url=URL('http://127.0.0.1:51119/api/compoundAnalyses/b9e0b3fd-17ab-46c7-b9e4-5e0ae40dc083/copy/notebook/')
```

So your response is HTML, but you are trying to interpret it as JSON.

**Comment by twentyse7en** [comment]
This is the response 
```
<ClientResponse(http://127.0.0.1:48891/api/compoundAnalyses/88393056-fbca-4b22-a549-0fa685974378/copy/notebook/) [404 Not Found]>
<CIMultiDictProxy('Server': 'TornadoServer/6.0.3', 'Content-Type': 'text/html; charset=UTF-8', 'Date': 'Thu, 06 Aug 2020 12:09:27 GMT', 'Content-Length': '69')>

```

File: src/libertem/web/notebook_generator/copy.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from .code_template import CodeTemplate
+
+
+def copyNotebook(conn, dataset, comp):
```
Comment (sk1p) [inline src/libertem/web/notebook_generator/copy.py:4]: Nitpick: function name should follow pep8 naming rules (`copy_notebook`)

File: client/src/compoundAnalysis/api.ts [hunk 2]
```diff
@@ -27,3 +27,13 @@ export async function createOrUpdateCompoundAnalysis(compoundAnalysisId: string,
 export async function removeCompoundAnalysis(compoundAnalysisId: string): Promise<RemoveCompoundAnalysisResponse> {
     return await genericDelete(`compoundAnalyses/${compoundAnalysisId}/`)
 }
+
+export async function getNotebook(compoundAnalysisId: string): Promise<CopyNotebookResponse>{
+    const basePath = getApiBasePath();
+    const url = `${basePath}compoundAnalyses/${compoundAnalysisId}/copy/notebook`;
```
Comment (sk1p) [inline client/src/compoundAnalysis/api.ts:33]: This may need a slash at the end of the URL

File: src/libertem/web/server.py [hunk 3]
```diff
@@ -72,6 +72,10 @@ def make_app(event_registry, shared_state):
             "state": shared_state,
             "event_registry": event_registry
         }),
+        (r"/api/compoundAnalyses/([^/]+)/copy/notebook", CopyScriptHandler, {
```
Comment (sk1p) [inline src/libertem/web/server.py:75]: May need a slash at the end

**Comment by sk1p** [comment]
I just tried the GUI functionality - really nice overall! What do you think about removing the "copy cells" button, and instead directly show the notebook cells and copy buttons in the "copy notebook" tab? Or is there a reason this button has to exist?

**Comment by twentyse7en** [comment]
> Or is there a reason this button has to exist?

No

> What do you think about removing the "copy cells" button, and instead directly show the notebook cells and copy buttons in the "copy notebook" tab?

Done

image

**Comment by twentyse7en** [comment]
If this design is okay, can I add the docs ?

**Comment by sk1p** [comment]
> > What do you think about removing the "copy cells" button, and instead directly show the notebook cells and copy buttons in the "copy notebook" tab?
> 
> Done

Great!

> If this design is okay, can I add the docs ?

Yes, looks good, go ahead!

**Comment by sk1p** [comment]
The AppVeyor failure appears to be unrelated, other than that it looks good so I'll merge. Thanks!

**Pull request merged**